code/modules/telesci/telesci_computer.dm

```diff
@@ -0,0 +1,309 @@
+	t += "<A href='?src=\ref[src];setangle=1'>Set Elevation</A>"
+	t += "<div class='statusDisplay'>[angle]�</div>"
+	t += "<span class='linkOn'>Set Power</span>"
+	t += "<div class='statusDisplay'>"
+
+	for(var/i = 1; i <= power_options.len; i++)
+		if(crystals.len < i)
+			t += "<span class='linkOff'>[power_options[i]]</span>"
+			continue
+		if(power == power_options[i])
+			t += "<span class='linkOn'>[power_options[i]]</span>"
+			continue
+		t += "<A href='?src=\ref[src];setpower=[i]'>[power_options[i]]</A>"
+
+	t += "</div>"
+	t += "<A href='?src=\ref[src];setz=1'>Set Sector</A>"
+	t += "<div class='statusDisplay'>[z_co ? z_co : "NULL"]</div>"
+
+	t += "<BR><A href='?src=\ref[src];send=1'>Send</A>"
+	t += " <A href='?src=\ref[src];receive=1'>Receive</A>"
+	t += "<BR><A href='?src=\ref[src];recal=1'>Recalibrate Crystals</A> <A href='?src=\ref[src];eject=1'>Eject Crystals</A>"
+
+	// Information about the last teleport
+	t += "<BR><div class='statusDisplay'>"
+	if(!last_tele_data)
+		t += "No teleport data found."
+	else
+		t += "Source Location: ([last_tele_data.src_x], [last_tele_data.src_y])<BR>"
+		//t += "Distance: [round(last_tele_data.distance, 0.1)]m<BR>"
+		t += "Time: [round(last_tele_data.time, 0.1)] secs<BR>"
+	t += "</div>"
+
+	var/datum/browser/popup = new(user, "telesci", name, 300, 500)
+	popup.set_content(t)
+	popup.open()
+	return
+
+/obj/machinery/computer/telescience/proc/sparks()
+	if(telepad)
+		var/datum/effect/effect/system/spark_spread/s = new /datum/effect/effect/system/spark_spread
+		s.set_up(5, 1, get_turf(telepad))
+		s.start()
+	else
+		return
+
+/obj/machinery/computer/telescience/proc/telefail()
+	sparks()
+	visible_message("<span class='warning'>The telepad weakly fizzles.</span>")
+	return
+
+/obj/machinery/computer/telescience/proc/doteleport(mob/user)
+
+	if(teleport_cooldown > world.time)
+		temp_msg = "Telepad is recharging power.<BR>Please wait [round((teleport_cooldown - world.time) / 10)] seconds."
+		return
+
+	if(teleporting)
+		temp_msg = "Telepad is in use.<BR>Please wait."
+		return
+
+	if(telepad)
+
+		var/truePower = Clamp(power + power_off, 1, 1000)
+		var/trueRotation = rotation + rotation_off
+		var/trueAngle = Clamp(angle, 1, 90)
+
+		var/datum/projectile_data/proj_data = projectile_trajectory(telepad.x, telepad.y, trueRotation, trueAngle, truePower)
+		last_tele_data = proj_data
+
+		var/trueX = Clamp(round(proj_data.dest_x, 1), 1, world.maxx)
+		var/trueY = Clamp(round(proj_data.dest_y, 1), 1, world.maxy)
+		var/spawn_time = round(proj_data.time) * 10
+
+		var/turf/target = locate(trueX, trueY, z_co)
+		var/area/A = get_area(target)
+		flick("pad-beam", telepad)
+
+		if(spawn_time > 15) // 1.5 seconds
+			playsound(telepad.loc, 'sound/weapons/flash.ogg', 25, 1)
+			// Wait depending on the time the projectile took to get there
+			teleporting = 1
+			temp_msg = "Powering up bluespace crystals.<BR>Please wait."
+
+
+		spawn(round(proj_data.time) * 10) // in seconds
```

Comment:
did you seriously just copy this archaic code from 2017 or something? dont use spawn and sleep